File: tests/TestAsset.rb
require "test/unit"
# FIXME: Remove this hardcoded path
require_relative "../build/generated/Asset"

class TestAsset < Test::Unit::TestCase
  def test_CDO_name
    assert_equal("CDO", CDO.new.name)
  end
  
  def test_CDO_PV
    assert_equal(CDO.new.PV, 0.0)
  end
end

File: tests/java/TestAsset.java
import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class TestAsset {
    @Test
    public void CDOhasPVofZero()
    {
        Asset.Asset cdo = new Asset.CDO();

        assertEquals(99.99, cdo.value(), 0.0);
    }

    @Test
    public void CDOisCalledCDO()
    {
        Asset.Asset cdo = new Asset.CDO();

        assertEquals(cdo.name(), "CDO");
    }
}
